src/core/mdn-annotation.js

```diff
@@ -0,0 +1,170 @@
+import { fetchAndCache } from "./utils";
+import hyperHTML from "hyperhtml";
+import mdnCss from "text!../../assets/mdn-annotation.css";
+
+export const name = "core/mdn-annoatation";
+
+const SPEC_MAP_URL =
+  "https://raw.githubusercontent.com/w3c/mdn-spec-links/master/SPECMAP.json";
+const JSON_BASE = "https://w3c.github.io/mdn-spec-links/";
+const MDN_URL_BASE = "https://developer.mozilla.org/en-US/docs/Web/";
+const MDN_BROWSERS = {
+  // The browser IDs here must match the ones in the imported JSON data.
+  // See the list of browser IDs at https://goo.gl/iDacWP.
+  chrome: "Chrome",
+  chrome_android: "Chrome Android",
+  edge: "Edge",
+  edge_mobile: "Edge Mobile",
+  firefox: "Firefox",
+  firefox_android: "Firefox Android",
+  ie: "Internet Explorer",
+  // nodejs: "Node.js", // no data for features in HTML
+  opera: "Opera",
+  opera_android: "Opera Android",
+  // qq_android: "QQ Browser", // not enough data for features in HTML
+  safari: "Safari",
+  safari_ios: "Safari iOS",
+  samsunginternet_android: "Samsung Internet",
+  // uc_android: "UC browser", // not enough data for features in HTML
+  // uc_chinese_android: "Chinese UC Browser", // not enough data for features in HTML
+  webview_android: "WebView Android",
+};
+
+async function fetchAndCacheJson(url, maxAge) {
+  if (!url) return {};
+  const request = new Request(url);
+  return await fetchAndCache(request, maxAge).then(r => r.json());
+}
+
+function insertMDNBox(node) {
+  let targetAncestor = node;
+  // Find the furthest ancestor that is a direct child of <body>
+  while (targetAncestor.parentNode.tagName !== "BODY") {
+    targetAncestor = targetAncestor.parentNode;
+  }
+  const targetSibling = targetAncestor.previousElementSibling;
+  if (targetSibling && targetSibling.classList.contains("mdn")) {
+    // If the target ancestor already has a mdnBox inserted, we just use it
+    return targetSibling;
+  }
+  const mdnBox = hyperHTML`<aside class="mdn before">
+    <input type="button" onclick="toggleStatus(this)" value="⋰" >
+  </aside>`;
+  document.body.insertBefore(mdnBox, targetAncestor);
+  return mdnBox;
+}
+
+function attachMDNDetail(container, mdnSpec) {
+  const { slug, summary } = mdnSpec;
+  container.innerHTML += `<b>MDN </b>`;
+  const mdnSubPath = slug.slice(slug.indexOf("/") + 1);
+  const mdnDetail = document.createElement("details");
+  mdnDetail.innerHTML += `
+    <summary>
+      <a title="${summary}" href="${MDN_URL_BASE}${slug}">${mdnSubPath}</a>
+    </summary>
+  `;
+  attachMDNBrowserSupport(mdnDetail, mdnSpec);
+  container.appendChild(mdnDetail);
+}
+
+function attachMDNBrowserSupport(container, mdnSpec) {
+  if (!mdnSpec.support) {
+    container.innerHTML += `<p class="nosupportdata">No support data.</p>`;
+    return;
+  }
+  const supportTable = hyperHTML`<p class="mdnsupport"></p>`;
```

Comment:
Was expecting something like:
```JS
const supportTable = hyperHTML`<p class="mdnsupport">${
  buildBrowserSupportTable()
}
</p>`;
```

Also... if it's a table... use a `<table>` instead of a paragraph and with spans 😉

Reply:
Yes, that's true. It's more like a table or a list. But that would also cause a lot of changes in the css file. Maybe it's simpler to keep it consistent with the HTML spec?

Reply:
heh, did they seriously use spans?

Reply:
Yes, the browser support list is a `<p>` containing containing `<span>`s for differents browsers. Also each browser `<span>` contains `<span>` for icon and support version.

Reply:
I guess if it's good enough for the html spec 🤷‍♂️